lib/pluginUtils.js

```diff
@@ -134,7 +135,14 @@ function registerPluginName (fn) {
   this[kRegisteredPlugins].push(name)
 }
 
+function checkPluginHealthiness (fn) {
+  if (fn.constructor.name === 'AsyncFunction' && fn.length === 3) {
+    warning.emit('FSTWRN002')
```

Comment:
I'm afraid this will trigger a lot of confusion without including the function name

You have spotted many wong plugins in out tests too - were them hard to find out?

Reply:
+1 on it. 
We should make a best effort. In case they submit an anonymous fn/plugin, maybe just a place instead?

Reply:
Sorry guys I'm not sure I understood what you mean. Do you mean to add a check if the function is anonymous and then check if is async and the number of parameters? @metcoder95 
Btw @Eomm fortunately it was easy to find them out

Reply:
I mean this use case:

you have an app with tons of plugins written by hundred of developers.
we ship this warning and your console shows a lot of messages, but you can't find the error because it can be hidden in the `node_modules`.

We have this util to get a function's name: 
- https://github.com/fastify/fastify/blob/main/lib/pluginUtils.js#L19
- https://github.com/fastify/fastify/blob/673a604ca70611fed177825dee932705e84b24a7/lib/pluginUtils.js#L146C3-L146C21

If we move this function call after the `registerPluginName.call(this, fn)`, it is possible to trigger a warning with a name, so it will be easier for the reader to understand which plugin or file triggers the warning.

Reply:
Didn't really think about it, you're right. I proposed a solution that slightly modify the `registerPluginName` behavior by registering also the anonymous plugin and from the unit tests everything seemed to be right. Let me know if this could break something or if the solution is fine. Later I also have to report these changes on #5141